Please address this warning, apart from that LGTM:
```
../librz/core/core.c: In function ‘rz_core_prompt’:
../librz/core/core.c:2812:3: error: implicit declaration of function ‘rz_core_gadget_print’; did you mean ‘rz_core_gadget_free’? [-Werror=implicit-function-declaration]
 2812 |   rz_core_gadget_print(r);
      |   ^~~~~~~~~~~~~~~~~~~~
      |   rz_core_gadget_free
cc1: all warnings being treated as errors
```

--- a/librz/core/cmd_print.c
+++ b/librz/core/cmd_print.c
@@ -1426,11 +1426,22 @@ static void cmd_print_fromage(RzCore *core, const char *input, const ut8 *data, 
 	}
 }
 
+/**
+ * \brief Frees a visual print gadget
+ *
+ * \param g reference to RzCoreGadget
+ */
 RZ_API void rz_core_gadget_free(RzCoreGadget *g) {
 	free(g->cmd);
 	free(g);
 }
 
+/**
+ * \brief Prints or displays the print gadgets while in
+ * visual mode
+ *
+ * \param core reference to RzCore
+ */
 RZ_API void rz_core_gadget_print(RzCore *core) {
 	RzCoreGadget *g;
 	RzListIter *iter;

--- a/librz/include/rz_core.h
+++ b/librz/include/rz_core.h
@@ -396,6 +396,9 @@ typedef int (*RzCoreSearchCallback)(RzCore *core, ut64 from, ut8 *buf, int len);
 #ifdef RZ_API
 RZ_API int rz_core_bind(RzCore *core, RzCoreBind *bnd);
 
+/**
+ * \brief APIs to handle Visual Gadgets
+ */
 RZ_API void rz_core_gadget_free(RzCoreGadget *g);
 RZ_API void rz_core_gadget_print(RzCore *core);
 

--- a/librz/include/rz_core.h
+++ b/librz/include/rz_core.h
@@ -397,6 +397,7 @@ typedef int (*RzCoreSearchCallback)(RzCore *core, ut64 from, ut8 *buf, int len);
 RZ_API int rz_core_bind(RzCore *core, RzCoreBind *bnd);
 
 RZ_API void rz_core_gadget_free(RzCoreGadget *g);
+RZ_API void rz_core_gadget_print(RzCore *core);
 
 RZ_API bool rz_core_plugin_init(RzCore *core);
 RZ_API bool rz_core_plugin_add(RzCore *core, RzCorePlugin *plugin);

--- a/test/db/cmd/cmd_visual
+++ b/test/db/cmd/cmd_visual
@@ -103,3 +103,25 @@ EXPECT=<<EOF
 0x00005b70  ffe0 660f 1f44 0000 c30f 1f80 0000 0000  ..f..D..........           [?25h[0m[2J[0;0H
 EOF
 RUN
+
+NAME=pg command print gadgets
+FILE=bins/elf/ls
+CMDS=<<EOF
+pg 6 6 35 35 il
+e scr.gadgets=1
+e scr.interactive=true
+e scr.columns=10
+e scr.rows=1
+< q; V
+?e
+EOF
+EXPECT=<<EOF
+[?25l[0;0H[0m[0x00005ae[0J[0m[s[6;0H[Link[7;0Hlibcap.[0m
+[8;0Hlibc.so[0m
+[9;0H[0m
+[10;0H2 libra[0m
+[11;0H[0m
+[12;0H[Linked[0m
+[0m[u[?25h[0m[2J[0;0H
+EOF
+RUN

